pkg/api/gitops/v1alpha1/application.go

```diff
@@ -17,12 +17,694 @@ limitations under the License.
 package v1alpha1
 
 import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
```

Comment:
Is it possible to import this structure from flux or helm-controller repo?

Reply:
Changing to latest version of controller-runtime seems to be a very important thing. 
Is there an updated detailed description of the work required for upgrading @kubesphere/sig-devops

Reply:
I prefer to do the upgrade work in a separate PR.

Reply:
we can use `github.com/fluxcd/helm-controller/api` now